workflows/rnaseq/rnaseq-standard-fastq.wdl

```diff
@@ -0,0 +1,123 @@
+##
+## ## LICENSING
+##
+## #### MIT License
+##
+## Copyright 2019 St. Jude Children's Research Hospital
+##
+## Permission is hereby granted, free of charge, to any person obtaining a copy of this
+## software and associated documentation files (the "Software"), to deal in the Software
+## without restriction, including without limitation the rights to use, copy, modify, merge,
+## publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons
+## to whom the Software is furnished to do so, subject to the following conditions:
+##
+## The above copyright notice and this permission notice shall be included in all copies or
+## substantial portions of the Software.
+##
+## THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+## BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+## NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+## DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+## OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+version 1.0
+
+import "https://raw.githubusercontent.com/stjudecloud/workflows/master/tools/fq.wdl"
+import "https://raw.githubusercontent.com/stjudecloud/workflows/master/tools/star.wdl"
+import "https://raw.githubusercontent.com/stjudecloud/workflows/master/tools/picard.wdl"
+import "https://raw.githubusercontent.com/stjudecloud/workflows/master/tools/ngsderive.wdl"
+import "https://raw.githubusercontent.com/stjudecloud/workflows/master/tools/htseq.wdl"
+import "https://raw.githubusercontent.com/stjudecloud/workflows/master/tools/samtools.wdl"
+import "https://raw.githubusercontent.com/stjudecloud/workflows/master/tools/util.wdl"
+import "https://raw.githubusercontent.com/stjudecloud/workflows/master/tools/deeptools.wdl"
+
+workflow rnaseq_standard {
+    input {
+        File gencode_gtf
+        Array[File] read_one_fastqs
+        Array[File] read_two_fastqs
+        File stardb_tar_gz
+        String strandedness = ""
+        String output_prefix = "sample"
+        String read_groups = ""
+        Int max_retries = 1
+        Boolean detect_nproc = false
+        Boolean validate_input = true
+    }
+
+    parameter_meta {
+        gencode_gtf: "GTF file provided by Gencode"
+        read_one_fastqs: "Input Fastq format file(s) with 1st read in pair to align"
+        read_two_fastqs: "Input Fastq format file(s) with 2nd read in pair to align"
+        stardb_tar_gz: "Database of reference files for the STAR aligner. Can be generated by `rnaseq-star-db-build.wdl`"
+        strandedness: "empty, 'Stranded-Reverse', 'Stranded-Forward', or 'Unstranded'. If missing, will be inferred"
+        output_prefix: "Prefix for output files"
+        max_retries: "Number of times to retry failed steps"
+        detect_nproc: "Use all available cores for multi-core steps"
+    }
```

Comment:
missing meta for `read_groups`

Reply:
The STAR step will actually fail without a well-formed `read_groups`. My `sort_star_input.py` script isn't skippable/doesn't allow an empty `read_groups` argument. I can go ahead and fix that if you want.

Reply:
Yes, thanks. I removed it for testing as the fastqs don't have read group information. I'll add it back.